### ISSUE TITLE
Intermittent: Cannot connect to the Docker daemon at unix:///var/run/docker.sock

### ISSUE THREAD

**Issue by anandt-ini8labs** [moby/moby#46874]
### Description

I am using docker on Centos for running my Kubernetes cluster.
I get intermittent error from Kubelet :
```
level=error msg="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running? 
```

### Reproduce

It is in intermittent issue

### Expected behavior

should not get the above error message

### docker version

```bash
Client: Docker Engine - Community
 Version:           19.03.15
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        99e3ed8919
 Built:             Sat Jan 30 03:17:57 2021
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.15
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       99e3ed8919
  Built:            Sat Jan 30 03:16:33 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.18
  GitCommit:        2456e983eb9e37e47538f59ea18f2043c9a73640
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

### docker info

```bash
Client:
 Debug Mode: false

Server:
 Containers: 134
  Running: 38
  Paused: 0
  Stopped: 96
 Images: 23
 Server Version: 19.03.15
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2456e983eb9e37e47538f59ea18f2043c9a73640
 runc version: v1.1.4-0-g5fd4c4d
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1160.71.1.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 32
 Total Memory: 62.78GiB
 ID: QHLD:FMNS:U36D:WV5Q:UBH5:H3IX:JFF2:FXSD:JPAH:43WA:YCAM:DO2C
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by thaJeztah** [comment]
It looks like you're running a very old version of the docker engine, which reached EOL 4 years ago. If there's a bug at hand, that won't be fixed for 19.03.

The error indicates that either the daemon stopped (hence the "is the daemon running?" in the error message), or for some reason, kubelet was not able to connect.

I would recommend;

- checking logs for the docker service, and your system logs (perhaps the daemon crashed for some reason?)
- if possible, and if you have a non-production environment to test on, update to a current version of docker (24.0.x is the currently maintained release), and a current version of containerd and runc


I'm closing this ticket because of the above, but feel free to continue the conversation and if there's information that may indicate there's a bug.

**Comment by anandt-ini8labs** [comment]
@thaJeztah Thanks for responding, the problem is it is intermittent and there are n logs in docker and and docker engine is also running so trying to understand what could be the possible cause of it.
